packages/patternfly-4/react-table/src/components/Table/Table.md

```diff
@@ -108,6 +108,110 @@ class SimpleTable extends React.Component {
 }
 ```
 
+## Reorderable Column Table
+
+```js
+import React from 'react';
+import {
+  Table,
+  TableHeader,
+  TableBody,
+  sortable,
+  SortByDirection,
+  headerCol,
+  TableVariant,
+  expandable,
+  cellWidth,
+  textCenter,
+} from '@patternfly/react-table';
+
+class ColReorderTable extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      columns: [
+        { title: 'Repositories' },
+        { title: 'Branches' },
+        { title: 'Pull requests' },
+        { title: 'Workspaces' },
+        {
+          title: 'Last Commit',
+          transforms: [textCenter],
+          cellTransforms: [textCenter]
+        }
+      ],
+      rows: [
+        {
+          cells: ['repositories', 'branches', 'PR', 'workspaces', 'last commit']
+        },
+        {
+          cells: [
+            {
+              title: <div>repositories - 2</div>,
+              props: { title: 'hover title' }
+            },
+            'branches - 2',
+            'PR - 2',
+            'workspaces - 2',
+            'last commit - 2'
+          ]
+        },
+        {
+          cells: [
+            'repositories - 3',
+            'branches - 3',
+            'PR - 3',
+            'workspaces - 3',
+            {
+              title: 'last commit - 3 (not centered)',
+              props: { textCenter: false }
+            }
+          ]
+        }
+      ]
+    };
+  }
+
+  render() {
+    const { columns, rows } = this.state;
+
+    const swapArrPosition = (from, to, arr) => {
+      let newArr = arr.slice();
+      [newArr[from], newArr[to]] = [newArr[to], newArr[from]];
+      return newArr;
+    }
+
+    const applyColumnReorder = (from, to) => {
```

Comment:
I know it's a pain to maintain, but I'd like us to take care of this problem on the PatternFly side, even if a wonky default implementation that can be overridden. We'll have to make a copy of the data or keep some state about the order of columns and iterate over them according to that.